File: ios/Lampada.m
//
//  Lampada.m
//  LightApp
//
//  Created by Ramon Cruz on 08/06/19.
//  Copyright © 2019 Facebook. All rights reserved.
//

#import <Foundation/Foundation.h>
// import RCTBridgeModule
#if __has_include(<React/RCTBridgeModule.h>)
#import <React/RCTBridgeModule.h>
#elif __has_include(“RCTBridgeModule.h”)
#import “RCTBridgeModule.h”
#else
#import “React/RCTBridgeModule.h” // Required when used as a Pod in a Swift project
#endif

@interface RCT_EXTERN_MODULE(Lampada, NSObject)
RCT_EXTERN_METHOD(turnOn)
RCT_EXTERN_METHOD(turnOff)
RCT_EXTERN_METHOD(getStatus: (RCTResponseSenderBlock)callback)

@end

File: ios/Lampada.swift
//
//  Lampada.swift
//  LightApp
//
//  Created by Ramon Cruz on 08/06/19.
//  Copyright © 2019 Facebook. All rights reserved.
//

import Foundation

@objc(Lampada)

class Lampada: NSObject{
  
  @objc
  static var isOn = false;
  
  @objc
  func turnOn(){
    Lampada.isOn = true;
    print("Lampada Ligada");
  }
  
  @objc
  func turnOff(){
    Lampada.isOn = false;
    print("Lampada Desligada");
  }
  
  @objc
  func getStatus(_ callback: RCTResponseSenderBlock){
    print("Lampada %@",Lampada.isOn);
    callback([NSNull(),Lampada.isOn])
    
  }
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return true;
  }
  
}
